tools/cli/commands/rsync.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import fs from 'fs';
+import process from 'process';
+import chalk from 'chalk';
+import execa from 'execa';
+import inquirer from 'inquirer';
+import watch from 'node-watch';
+import tmp from 'tmp';
+/**
+ * Internal dependencies
+ */
+import { projectDir } from '../helpers/install.js';
+import { listProjectFiles } from '../helpers/list-project-files.js';
+import { allProjectsByType, dirs } from '../helpers/projectHelpers.js';
+import { runCommand } from '../helpers/runCommand.js';
+
+/**
+ * Entry point for the CLI.
+ *
+ * @param {object} argv - The argv for the command line.
+ */
+export async function rsyncInit( argv ) {
+	argv = await promptForPlugin( argv );
+	argv = await promptForDest( argv );
+	const sourcePluginPath = projectDir( `plugins/${ argv.plugin }` );
+
+	if ( true === argv.watch ) {
+		console.log( chalk.yellow( `Watching for changes in ${ argv.plugin } to rsync` ) );
+		watch( sourcePluginPath, { recursive: true }, ( eventType, fileName ) => {
+			if ( fileName && 'update' === eventType ) {
+				if ( argv.v ) {
+					console.log( chalk.yellow( `Changed detected in ${ fileName }...` ) );
+				}
+				rsyncToDest( sourcePluginPath, argv.dest );
+			}
+		} );
+	} else {
+		await rsyncToDest( sourcePluginPath, argv.dest );
+	}
+}
+
+/**
+ * Function that does the actual work of rsync.
+ *
+ * @param {string} source - Source path.
+ * @param {string} dest - Destination path.
+ * @returns {Promise<void>}
+ */
+async function rsyncToDest( source, dest ) {
+	let includeFiles = [];
+	for await ( const projectFiles of listProjectFiles( source, execa ) ) {
+		includeFiles.push( projectFiles );
+	}
+	includeFiles = includeFiles.map(
+		v => '/' + ( v.match( /[[*?]/ ) ? v.replace( /[[*?\\]/g, '\\$&' ) : v )
+	);
```

Comment:
I suggested this, and reviewing it now I was wondering if it was wrong or overcomplicated. 😀 That probably means it needs a comment!
```suggestion
	// rsync looks for `[`, `*`, or `?` to decide if it's a literal string or a pattern.
	// Only patterns need backslash escapes, literal strings must not have them.
	includeFiles = includeFiles.map(
		v => '/' + ( v.match( /[[*?]/ ) ? v.replace( /[[*?\\]/g, '\\$&' ) : v )
	);
```

Reply:
😄